src/components/RadioInput/RadioInput.js

```diff
@@ -0,0 +1,167 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+import { hideVisually } from 'polished';
+import { shadowSingle, disableVisually } from '../../styles/style-helpers';
+
+const baseStyles = ({ theme }) => css`
+  label: radio-input__label;
+  color: ${theme.colors.n700};
+  padding-left: ${theme.spacings.giga};
+  position: relative;
+
+  &::before {
+    ${shadowSingle({ theme })};
+    background-color: ${theme.colors.white};
+    border: 1px solid ${theme.colors.n500};
+    border-radius: 100%;
+    content: '';
+    display: block;
+    height: ${theme.spacings.mega};
+    position: absolute;
+    top: 50%;
+    left: 0;
+    transform: translateY(-50%);
+    width: ${theme.spacings.mega};
+    transition: border 0.05s ease-in;
+  }
+
+  &::after {
+    background-color: ${theme.colors.b500};
+    border-radius: 100%;
+    content: '';
+    display: block;
+    position: absolute;
+    top: 50%;
+    left: ${theme.spacings.bit};
+    height: ${theme.spacings.byte};
+    width: ${theme.spacings.byte};
+    transform: translateY(-50%) scale(0, 0);
+    visibility: hidden;
+    transition: transform 0.05s ease-in, visibility 0.05s ease-in;
+  }
+`;
+
+const checkedStyles = ({ theme, checked }) =>
+  checked &&
+  css`
+    label: radio-input--active;
+
+    &::before {
+      border-color: ${theme.colors.b500};
+    }
+
+    &::after {
+      transform: translateY(-50%) scale(1, 1);
+      visibility: visible;
+    }
+  `;
+
+const invalidStyles = ({ theme, isInvalid }) =>
+  isInvalid &&
+  css`
+    label: radio-input--error;
+    &:not(:focus)::before {
+      background-color: ${theme.colors.r100};
+      border-color: ${theme.colors.r300};
+    }
+
+    &:not(:focus)::after {
+      background-color: ${theme.colors.r100};
+    }
+  `;
+
+const disabledStyles = ({ theme, disabled }) =>
+  disabled &&
+  css`
+    label: radio-input--disabled;
+    ${disableVisually()};
+
+    &::before {
+      border-color: ${theme.colors.n500};
+    }
+
+    &::after {
+      background-color: ${theme.colors.n500};
+    }
+  `;
+
+const inputStyles = ({ theme }) => css`
+  label: radio-input__input;
+  ${hideVisually()};
+
+  &:focus + label::before {
+    border-width: 2px;
+    border-color: ${theme.colors.b500};
+  }
+`;
+
+const Input = styled('input', { label: 'RadioInputInput' })(inputStyles);
+
+const Label = styled('label', { label: 'RadioInputLabel' })(
+  baseStyles,
+  checkedStyles,
+  disabledStyles,
+  invalidStyles
+);
+
+/**
+ * RadioInput component for forms.
+ */
+const RadioInput = ({ onClick, label, name, ...props }) => (
+  <Fragment>
+    <Input id={name} type="radio" {...{ ...props, name }} />
```

Comment:
As with Felix's implementation for the selects, I think we're a bit too far away from the DOM for such a low-level component.

I don't think we want to make assumptions about the label if we can help it. For instance, assuming the label is going to be a string - it could very well be a component, imagine a label containing a price, e.g. `My stuff costs <Price number="15" currency="EUR"/>`.

Also, that we may end up with props that simply cannot be forwarded to both components.

I would go for using `{children}` here and let the end user use the `<Label>` component themselves.

For instance:

```
<RadioInput name="my-option">
  <Label htmlFor="my-option">This is my <Price amount="16"/></Label>
</RadioInput>
```

But now that I think about this - and the fact that we're using all this fancy css-in-js stuff -- do we actually even need a separate component for radios and checkboxes? Can't we just take the "type" prop and determine the correct styles?

In that way we would stay super close to the DOM and give the end user ultimate flexibility. WDYT? + @felixjung